viewing_party/party.py

```diff
@@ -1,23 +1,160 @@
-# ------------- WAVE 1 --------------------
+import statistics
+from statistics import mode
 
+# -----------------------------------------
+# ------------- WAVE 1 --------------------
+# -----------------------------------------
 def create_movie(title, genre, rating):
-    pass
+    """Creates a movie dict."""
+    if not title or not genre or not rating:
+        return None
+    return {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+
+def add_to_watched(user_data, movie):
+    """Adds movie to a user's watched list."""
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """Adds a movie to a user's watchlist."""
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie):
+    """Moves a movie from a user's watch list to their watched list."""
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == movie:
+            movie_dict = user_data["watchlist"].pop(i)
+            user_data["watched"].append(movie_dict)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    """Gets the average rating of movies a user has watched."""
+    ratings = [] 
+    watched_list = user_data["watched"]
+    
+    for movie in watched_list:
+        ratings.append(movie["rating"]) 
 
+    ratings_sum = 0.0
+    for rating in ratings:
+        ratings_sum += rating
 
-# -----------------------------------------
+    try:
+        avg_rating = ratings_sum / len(ratings)
+    except ZeroDivisionError:
+        return 0.0
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    """Finds a user's most watched genre."""
+    genres = [] 
+    watched_list = user_data["watched"] 
+
+    for movie in watched_list:
+        genres.append(movie["genre"])
+
+    if genres:
+        return mode(genres)

```

Comment:
👀 I'm so intrigued by this implementation! It looks so clean and efficient -- that said, I'm also curious to hear how you would implement this by hand if you didn't have access to the statistics module!